File: src/main/java/com/handwoong/everyonewaiter/service/WaitingService.java
package com.handwoong.everyonewaiter.service;

import com.handwoong.everyonewaiter.dto.WaitingDto;
import java.util.List;
import java.util.UUID;

public interface WaitingService {

    WaitingDto.CountResponseDto count(String username, Long storeId);

    WaitingDto.ResponseDto findWaiting(UUID waitingId);

    List<WaitingDto.ResponseDto> findDefaultWaitingList(String username, Long storeId);

    UUID register(String username, Long storeId, WaitingDto.RequestDto waitingDto);

    void enterWaiting(String username, Long storeId, UUID waitingId);

    void cancelWaiting(UUID waitingId);
}

File: src/main/kotlin/com/handwoong/everyonewaiter/service/waiting/WaitingService.kt
package com.handwoong.everyonewaiter.service.waiting

import com.handwoong.everyonewaiter.dto.waiting.WaitingCountResponse
import com.handwoong.everyonewaiter.dto.waiting.WaitingRegisterRequest
import com.handwoong.everyonewaiter.dto.waiting.WaitingResponse
import java.util.*

interface WaitingService {

    fun count(username: String, storeId: Long): WaitingCountResponse

    fun findWaiting(waitingId: UUID): WaitingResponse

    fun findStatusWaitWaitingList(username: String, storeId: Long): List<WaitingResponse>

    fun register(username: String, storeId: Long, waitingDto: WaitingRegisterRequest): Int

    fun enterWaiting(username: String, storeId: Long, waitingId: UUID)

    fun cancelWaiting(storeId: Long, waitingId: UUID)

}
